pkg/controller/controller.go

```diff
@@ -248,8 +248,7 @@ func (ctrl *resizeController) Run(
 	}
 
 	if !cache.WaitForCacheSync(stopCh, informersSyncd...) {
-		klog.Errorf("Cannot sync pod, pv or pvc caches")
-		return
+		klog.Fatalf("Cannot sync pod, pv or pvc caches")
```

Comment:
If it calls os.Exit here, does that mean line 239 `defer ctrl.claimQueue.ShutDown()` will be skipped?  It seems that line 242 is skipped.

Reply:
So I think we still need to handle ctrl.claimQueue.ShutDown() as os.Exit is called by klog.Fatalf.
https://godoc.org/github.com/kubernetes/klog#Fatalf

Reply:
yeah makes sense. What about returning an error from `Run` which can be caught by the caller?

Reply:
What happens if we don't shutdown the queue before exiting? It is just internal cached state anyways and if a process dies  then so does its internal state.